tests/install_upgrade_operators/network_policy/conftest.py

```diff
@@ -0,0 +1,186 @@
+    service_ip = service_by_network_policy.instance.spec.get("clusterIP")
+    assert service_ip, f"Service {service_by_network_policy.name} has no clusterIP"
+    if ":" in service_ip:
+        pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
+    return service_ip
+
+
+@pytest.fixture
+def service_port(service_by_network_policy):
+    ports = service_by_network_policy.instance.spec.get("ports", [])
+    assert len(ports) > 0, f"Service {service_by_network_policy.name} has no ports defined"
+
+    service_port = ports[0].get("port")
+    assert service_port, f"Service {service_by_network_policy.name} first port has no port number"
+    return service_port
+
+
+@pytest.fixture
+def pods_by_network_policy(admin_client, hco_namespace, network_policy_by_name, network_policy_match_labels):
+    label_pairs = [f"{key}={value}" for key, value in network_policy_match_labels.items()]
+    full_label_selector = ",".join(label_pairs)
+    pods = Pod.get(dyn_client=admin_client, namespace=hco_namespace.name, label_selector=full_label_selector)
+    pod_names = [pod.name for pod in pods if pod.status == Pod.Status.RUNNING]
+    assert len(pod_names) > 0, (
+        f"No running pods found matching NetworkPolicy {network_policy_by_name.name} podSelector "
+        f"in namespace {hco_namespace.name}"
+    )
+    pod_names.sort()
+    return pod_names
+
+
+@pytest.fixture
+def ingress_connectivity_baseline(
+    admin_client, hco_namespace, deployed_client_pod, service_ip, service_port, network_policy_match_labels
+):
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        try:
+            deployed_client_pod.execute(
+                command=shlex.split(f"curl -sS -k --connect-timeout 5 https://{service_ip}:{service_port}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Ingress connectivity test preparation FAILED: With allow-all NetworkPolicy, "
+                f"connection from test client pod to the CNV service should succeed but got error: {e}"
+            )
+    yield {
+        "deployed_client_pod": deployed_client_pod,
+        "service_ip": service_ip,
+        "service_port": service_port,
+    }
+
+
+@pytest.fixture
+def egress_connectivity_baseline(
+    admin_client, hco_namespace, pods_by_network_policy, network_policy_match_labels, deployed_server_service
+):
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        pod_name = pods_by_network_policy[0]
+        service_ip = deployed_server_service.instance.spec.get("clusterIP")
+        if ":" in service_ip:
+            pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
```

Comment:
you call `deployed_server_service.instance.spec.get("clusterIP")` here and in the test itself, create the fixture and reuse